test/grammar/left_recursion_test.py

```diff
@@ -310,8 +307,8 @@ def test_left_recursion_bug(self, trace=False):
             start = expression $ ;
 
             expression =
-                | paren_expression
```

Comment:
It doesn't consider `minus_expression` a valid alternative after it successfully parsed a `paren_expression`, 
therefore `(3 - 2) - 1` was correctly failing.